include/aws/auth/signing_config.h

```diff
@@ -163,28 +163,36 @@ struct aws_signing_config_aws {
     struct aws_date_time date;
 
     /**
-     * Optional function to control which parameters (header or query) are a part of the canonical request.
-     * Skipping auth-required params
-     * will result in an unusable signature.  Headers injected by the signing process are not skippable.
+     * Optional function to control which headers are a part of the canonical request.
+     * Skipping auth-required headers will result in an unusable signature.  Headers injected by the signing process
+     * are not skippable.
      *
      * This function does not override the internal check function (x-amzn-trace-id, user-agent), but rather
      * supplements it.  In particular, a header will get signed if and only if it returns true to both
      * the internal check (skips x-amzn-trace-id, user-agent) and this function (if defined).
      */
-    aws_should_sign_param_fn *should_sign_param;
-    void *should_sign_param_ud;
-
-    /**
-     * We assume the uri will be encoded once in preparation for transmission.  Certain services
-     * do not decode before checking signature, requiring us to actually double-encode the uri in the canonical request
-     * in order to pass a signature check.
-     */
-    bool use_double_uri_encode;
-
-    /**
-     * Controls whether or not the uri paths should be normalized when building the canonical request
-     */
-    bool should_normalize_uri_path;
+    aws_should_sign_header_fn *should_sign_header;
+    void *should_sign_header_ud;
+
+    struct {
+        /**
+         * We assume the uri will be encoded once in preparation for transmission.  Certain services
+         * do not decode before checking signature, requiring us to actually double-encode the uri in the canonical
+         * request in order to pass a signature check.
+         */
+        uint32_t use_double_uri_encode : 1;
```

Comment:
not sure it's worth the effort here, changing the types and moving them in a sub-struct.
You could keep the external API unchanged and get the same size-saving effect if you just modified them like:
`bool use_double_uri_encode : 1;`